server/src/UseCases/Games/CalculateScore/CalculateScoreCommandHandler.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Autofac.Features.Indexed;
+using BGS.ApplicationCore.Entities;
+using BGS.ApplicationCore.Games.Interfaces;
+using BGS.ApplicationCore.Games.Specifications;
+using BGS.SharedKernel;
+using MediatR;
+
+namespace BGS.UseCases.Games.CalculateScore;
+
+internal class CalculateScoreCommandHandler : IRequestHandler<CalculateScoreCommand, int>
+{
+    private readonly IIndex<string, IGameScoringStrategy> _gameScoringStrategy;
+    private readonly IRepository<Game> _gameRepository;
+
+    public CalculateScoreCommandHandler(
+        IIndex<string, IGameScoringStrategy> gameScoringStrategy,
+        IRepository<Game> gameRepository)
+    {
+        _gameScoringStrategy = gameScoringStrategy;
+        _gameRepository = gameRepository;
+    }
+
+    public async Task<int> Handle(CalculateScoreCommand command, CancellationToken cancellationToken)
+    {
+        var gameKey = await _gameRepository.SingleOrDefaultAsync(
+            new GameKeyByIdSpecification(command.GameId), cancellationToken);
+
+        return _gameScoringStrategy[gameKey].Calculate(command.GameData);
```

Comment:
I prefer to take this logic into the Service and test it

Reply:
The Use Cases (or Application Services) project is a relatively thin layer that wraps the domain model and are typically organized by feature. These may be simple CRUD operations or much more complex activities. Use Cases are often grouped into Commands and Queries, following CQRS.